docs/container-measurement-design.md

```diff
@@ -164,6 +164,39 @@ Within the measurement list, system processes can be explicitly defined and meas
+
+### Remote Attestation
+
+Remote attestation is a process where a remote party verifies the integrity and trustworthiness of a system. This can be achieved using several types of evidence, including:
+
+* Signed Report: This report, such as a Quote in 
+TDX or TPM, contains the platform's Trusted Computing Base (TCB) and the runtime measurement register. The TCB includes the hardware and software components critical for the system’s security. The signed report ensures that the measurements have been recorded correctly and have not been altered.
+
+    * Signer Verification: Before trusting the report, it is crucial to verify the identity of the signer. This step ensures that the report comes from a legitimate source and has not been tampered with.
+
+    * Measurement Register: The measurement register, which logs the state of the system components, can be trusted if the signer is verified. This register includes measurements of the firmware, boot loader, operating system, and applications, ensuring the integrity of the entire system.
+
+* Nonce: A nonce is a random number used once in cryptographic communication to ensure that old communications cannot be reused in replay attacks. It ensures the freshness of the attestation evidence.
+
+* Event Logs: Event logs record significant events within the system, such as changes in configuration or application execution. By replaying these logs and comparing the results with the measurement registers, we can verify that the logs accurately reflect the system’s state. If the replayed logs match the measurement registers, it confirms that the logs are trustworthy.
+
+    * Reconstruction of Environment: Using the event logs, we can reconstruct the entire virtual machine (VM) environment. This reconstruction includes measurements for firmware, GRUB, shim, kernel, initrd, and all applications inside the container. This process helps ensure that no unauthorized changes have occurred.
+* User Data: User data can also be part of the attestation process to ensure that sensitive information has not been compromised.
+
+A third-party application can maintain the golden values (trusted baseline measurements) for all these components and use them to verify the attestation evidence. This verification ensures that the system is in a known good state before sensitive data is processed or transmitted.
+
+### Local Attestation
+Local attestation is similar to remote attestation but occurs between applications running on the same local system. It is used when two or more local applications need to verify each other’s integrity without involving a third party. This is useful in scenarios where local components must ensure mutual trust before performing sensitive operations.
```

Comment:
The same as above. Any reference for the definition and procedure of local attestation?